File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1
        right = max(piles)
        while left <= right:
            m = (left + right) // 2
            total = 0
            for num in piles:
                if num % m == 0:
                    total += num // m
                else:
                    total += (num // m) + 1
            if total <= h:
                right = m - 1
            else:
                left = m + 1
            
        return left

File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0, right = heights.length - 1, max = 0, stop = 0;
        while(left < right) {
            stop = (heights[left] <= heights[right]) ? heights[left] : heights[right];
            if ((stop * (right - left)) > max) {
                max = stop * (right - left);
            }
            if (heights[left] <= heights[right]) {
                left += 1;
            } else {
                right -= 1;
            }
        }
        return max;
    }
}
